client/my-sites/woocommerce/woop/landing-page.tsx

```diff
@@ -1,81 +1,57 @@
+import useWooCommerceOnPlansEligibility from 'calypso/signup/steps/woocommerce-install/hooks/use-woop-handling';
+import isAtomicSite from 'calypso/state/selectors/is-site-automated-transfer';
+
+import './style.scss';
 
 interface Props {
 	startSetup: () => void;
-	isFeatureActive: boolean;
-	upgradingPlan: Record< string, unknown >;
-	siteSlug: string;
+	siteId: number;
 }
 
 const images = [ { src: Image01 }, { src: Image02 }, { src: Image03 }, { src: Image04 } ];
 
-const WoopLandingPage: React.FunctionComponent< Props > = ( {
-	startSetup,
-	siteSlug,
-	isFeatureActive,
-	upgradingPlan,
-} ) => {
+const WoopLandingPage: React.FunctionComponent< Props > = ( { startSetup, siteId } ) => {
 	const { __ } = useI18n();
 	const navigationItems = [ { label: 'WooCommerce', href: `/woocommerce-installation` } ];
 	const ctaRef = useRef( null );
 
-	function onCTAHandler() {
-		if ( ! isFeatureActive && upgradingPlan?.product_slug ) {
-			return ( window.location.href = addQueryArgs(
-				{ redirect_to: window.location.href },
-				`/checkout/${ siteSlug }/${ upgradingPlan.product_slug }`
-			) );
-		}
-
-		startSetup();
-	}
-
-	const upgradeButtonText = upgradingPlan.product_name
-		? sprintf(
-				/* translators: %s: The upgrading plan name (ex.: WordPress.com Business) */
-				__( 'Upgrade to %s' ),
-				upgradingPlan.product_name
-		  )
-		: __( 'Upgrade' );
+	const { hasBlockers, wpcomDomain } = useWooCommerceOnPlansEligibility( siteId );
+	const isAtomic = !! useSelector( ( state ) => isAtomicSite( state, siteId ) );
 
-	let headlineText;
+	function onCTAClickHandler() {
+		if ( isEnabled( 'woop' ) && ! isAtomic ) {
+			return page( `/start/woocommerce-install/?site=${ wpcomDomain }` );
+		}
 
-	if ( isFeatureActive ) {
-		headlineText = __( 'Build exactly the eCommerce website you want.' );
-	} else {
-		headlineText = upgradingPlan.product_name
-			? sprintf(
-					/* translators: %s: The upgrading plan name (ex.: WordPress.com Business) */
-					__( 'Upgrade to the %s plan and set up your WooCommerce store.' ),
-					upgradingPlan.product_name
-			  )
-			: __( 'Upgrade to set up your WooCommerce store.' );
+		return startSetup();
 	}
 
 	return (
 		<div className="woop__landing-page">
 			<FixedNavigationHeader navigationItems={ navigationItems } contentRef={ ctaRef }>
-				<Button onClick={ onCTAHandler } primary>
-					{ isFeatureActive ? __( 'Set up my store!' ) : upgradeButtonText }
+				<Button onClick={ onCTAClickHandler } primary disabled={ hasBlockers }>
+					{ __( 'Set up my store!' ) }
 				</Button>
 			</FixedNavigationHeader>
 			<CtaSection
 				title={ __( 'Have something to sell?' ) }
-				headline={ headlineText }
-				buttonText={ isFeatureActive ? __( 'Set up my store!' ) : upgradeButtonText }
-				buttonAction={ onCTAHandler }
+				headline={ __( 'Build exactly the eCommerce website you want.' ) }
+				buttonText={ __( 'Set up my store!' ) }
+				buttonAction={ onCTAClickHandler }
+				buttonDisabled={ hasBlockers }
```

Comment:
```suggestion
				buttonDisabled={ isFetching || hasBlockers } // note requires isFetching to be defined above
```

We should do something like this, not a blocker for now. I tried it out and I think we need to do other work on the status requests before we can do this.